doc/dev_guide/internals/msgpack_extensions.rst

```diff
@@ -233,3 +240,141 @@ the server response will look like this:
             00                              MP_UINT = error number
             05                              MP_ERROR_ERRCODE
             0a                              MP_UINT = eror code ER_SPACE_EXISTS
+
+
+..  _msgpack_ext-datetime:
+
+**********************************
+The DATETIME type
+**********************************
+
+Since version :doc:`2.10.0 </release/2.10.0>`.
+The MessagePack EXT type ``MP_EXT`` together with the extension type
+``MP_DATETIME`` is a header for values of the DATETIME type.
+It creates a container of 8 or 16 bytes long payload.
+
+``MP_DATETIME`` type is 4.
+
+The `MessagePack spec <https://github.com/msgpack/msgpack/blob/master/spec.md>`_
+defines ``d7``to mean ``fixext`` with size 8 or ``d8`` to mean ``fixext`` with size 16.
+
+So the datetime MessagePack representation looks like this:
+
+..  code-block:: none
+
+    +---------+----------------+==========+-----------------+
+    | MP_EXT  | MP_DATETIME    | seconds  | nsec; tzoffset; |
+    | = d7/d8 | = 4            |          | tzindex;        |
+    +---------+----------------+==========+-----------------+
+
+MessagePack data contains:
+
+*   Seconds (8b) as full, unencoded, signed 64-bit integer, stored in little-endian order.
+
+*   The optional fields (8b), if any of them have a non-zero value.
+    The fields include nsec, tzoffset, and tzindex, packed in little-endian order.
+
+For more information about datetime, see :ref:`datetime field type details <index-box_datetime>`
+and :doc:`module datetime </reference/reference_lua/datetime>`.
```

Comment:
> and :doc:`module datetime </reference/reference_lua/datetime>`

Better to say, "description of the datetime module" or "reference for the datetime module".